File: 283-move-zeroes/283-move-zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    // let count = 0;
    // for(let i = 0; i < nums.length; i++)
    //     if(nums[i] === 0)
    //         count++;
    // let ans = [];
    // for(let i = 0; i < nums.length; i++)
    //     if(nums[i]) ans.push(nums[i]);
    // while(count--)
    //     ans.push(0);
    // for(let i = 0; i < nums.length; i++)
    //     nums[i] = ans[i];
    let countZero = 0;
    for(let  i = 0; i < nums.length; i++)
    {
        if(nums[i] != 0)
            nums[countZero++] = nums[i];
    }
    for(let i = countZero; i < nums.length; i++)
        nums[i] = 0;
};

File: 283-move-zeroes/283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int countZero = 0;
        for(int i = 0; i < nums.size(); i++)
        {
            if(nums[i]!=0)
            {
                nums[countZero] = nums[i];
                countZero++;
            }
        }
        for(int i = countZero; i < nums.size(); i++)
            nums[i] = 0;
    }
};